### PR TITLE
Refactor/login tests and feat/user dashboard

### LINKED ISSUE #19: User dashboard
User Story

When I log in
I am taken to my user dashboard page
Where I see a button to "Search for new recipes" that takes me to "dashboard/search"
I no longer the see the "Login" button on my navbar
I now see a button to redirect me to my dashboard permanently on the navbar
I see a message that tells me I am currently logged in as myself

For "My Recipes":
I see a list of my recipes with their name, a small Icon
Each name is a link to the recipes show
Each recipe is listed in order of most recently added

### PR DESCRIPTION
**Pull request by lanebret85**
Description:
- added tests and code for user dashboard page and logout functionality
- fixed errors in test and code from login functionality originating from original code that created/stored sessions in the backend app

Necessary checkmarks:
- [x] All Tests are Passing
- [x] The code will run locally

Type of change
- [x] New feature
- [x] Test
- [x] Refactor
- [ ] Bug Fix
- [ ] Documentation
      
Implements/Fixes:
    description closes #19 
    
Testing Changes
- [ ] No Tests have been changed
- [x] Some Tests have been changed
- [ ] All of the Tests have been changed(Please describe what in the world happened)
      
Checklist:
- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [x] I have fully tested my code

(For Fun!) Please Include a link to a gif of your feelings about this branch
Link:

### CODE REVIEW

**Review by JorjaF — APPROVED** [review]
looks great, thank you for all your hard work

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -9,10 +9,22 @@
     <%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>
     <%= javascript_importmap_tags %>
 
-    <%= button_to "Login", "/login", method: :get %>
+    <% if current_user %>
```
Comment (Antoine-Aube) [inline app/views/layouts/application.html.erb:12]: oh was it in order to be able to use it in the views? interesting.

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -9,10 +9,22 @@
...
+      <%= button_to "My Dashboard", "/dashboard", method: :get %>
+      <%= button_to "Log out", "/logout", method: :get %>
+    <% else %>
+      <% unless controller_name == 'sessions' && action_name == 'login_form' %>
```
Comment (Antoine-Aube) [inline app/views/layouts/application.html.erb:16]: interesting syntax, never seen that before.

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -10,11 +11,16 @@ def login
       session[:user_id] = response[:user_id]
       redirect_to "/dashboard"
     else
-      flash[:error] = "Invalid credentials"
+      flash[:error] = response[:error]
```
Comment (Antoine-Aube) [inline app/controllers/sessions_controller.rb:14]: Also curious what prompted the change here. just wanting the error message from the json response?

**Review by Antoine-Aube — APPROVED** [review]

**Pull request merged**